File: 1403. Minimum Subsequence in Non-Increasing Order/Solution.py
from typing import List


class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        sum1, sum2 = 0, sum(nums)
        res = []
        for x in sorted(nums, reverse=True):
            res.append(x)
            sum1 += x
            sum2 -= x
            if sum1 > sum2:
                break
        return res

File: 1403. Minimum Subsequence in Non-Increasing Order/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        int sum1 = 0, sum2 = 0;
        int[] hash = new int[101];
        for (int x : nums) {
            hash[x]++;
            sum2 += x;
        }
        List<Integer> res = new ArrayList<>();
        for (int i = 100; i > 0; i--) {
            while (hash[i]-- > 0) {
                res.add(i);
                sum1 += i;
                sum2 -= i;
                if (sum1 > sum2) return res;
            }
        }
        return null;
    }
}
